tests/test_riscv_printer.py

```diff
@@ -0,0 +1,53 @@
+from io import StringIO
+from typing import Sequence
+
+from xdsl.ir import OpResult
+from xdsl.riscv_asm_writer import print_riscv_module
+from xdsl.dialects import riscv, builtin
+from xdsl.irdl import irdl_op_definition, IRDLOperation, Annotated, OpAttr
+from xdsl.ir import SSAValue
+from xdsl.dialects.builtin import IntegerType, IndexType, IntegerAttr
+from xdsl.builder import Builder
+
+
+@irdl_op_definition
+class ExternalRiscvOp(IRDLOperation, riscv.RISCVPrinterInterface):
+    name = "custom.custom_op"
```

Comment:
Why not add this to the riscv dialect?

Reply:
I wanted to showcase that ops that are not part of the base risc-v dialect can be printed to risc-v if they so desire

Reply:
I see. I guess this would be useful for dialects that are extensions to RISC-V... Would be good to sit down and slice apart the riscv dialect at some point